File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/Machine.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import Machine._

case class Machine (
  /* ID of the machine. */
  id: Long,
/* Name of the machine. */
  name: Option[String],
/* Notes about the machine */
  notes: Option[String])

object Machine {
  import DateTimeCodecs._

  implicit val MachineCodecJson: CodecJson[Machine] = CodecJson.derive[Machine]
  implicit val MachineDecoder: EntityDecoder[Machine] = jsonOf[Machine]
  implicit val MachineEncoder: EntityEncoder[Machine] = jsonEncoderOf[Machine]
}

File: openapi-generator/javascript-closure-angular/API/Client/Machine.js
goog.provide('API.Client.Machine');

/**
 * Contains information about a machine.
 * @record
 */
API.Client.Machine = function() {}

/**
 * ID of the machine.
 * @type {!number}
 * @export
 */
API.Client.Machine.prototype.id;

/**
 * Name of the machine.
 * @type {!string}
 * @export
 */
API.Client.Machine.prototype.name;

/**
 * Notes about the machine
 * @type {!string}
 * @export
 */
API.Client.Machine.prototype.notes;

